timemachine/fe/dummy.py

```diff
@@ -0,0 +1,396 @@
+import numpy as np
+import networkx as nx
+import copy
+import itertools
+
+
+def _add_successors(mol, core, groups):
```

Comment:
Observation: The definitions in this file seem purely graph-based, but the signatures are in terms of full-fledged RDKit mols.

Question: Would it be preferable if these accepted networkx graphs instead?

------

(Primary consideration: Unsure if there are important logical differences in the graph traversal procedures here vs. [in networkx](https://networkx.org/documentation/stable/reference/algorithms/traversal.html#module-networkx.algorithms.traversal.breadth_first_search), aside from the adaptations to use RDKit molecules in the signatures.

Secondary consideration: Implementations might be marginally cleaner if iteration over nodes, edges, neighborhoods, etc. don't require `GetAtomWithIdx()`, `GetIdx()`, `GetBeginAtomIdx()` etc.)

Reply:
I debated about this too - I wrote this initially in rdkit simply because

1) I'm extremely familiar with the traversal/query logic, though there's no real barrier to adapting this to networkx

2) There was another version of the code from before that predicated on atom/bond properties ("eg. heavy atom or not" , "is SP3 hybridized", "single or double bond" etc.). Although the code currently doesn't predicate on more tricky logic, there may be instances where we need to avoid certain types of torsions defined over core anchors such as nitriles (where the collinear bonds would vanish the cross product).

Reply:
Just wanted to +1 the idea of separating the graph traversal logic from the details of the RDKit data structure and potentially using `bfs_successors`

Reply:
This turned out to be doable, but more complicated than I thought, mainly because of the need to deal with avoiding back-edges (since edges are undirected, a successor could actually point back to a predecessor if I'm not super careful). I'd still be happy to convert this to networkx but I'd rather save that for another PR (since there are probably other parts of this code that could benefit from the switch to networkx).

Reply:
After discussing with @mcwitt we think its best to defer the nx conversion stuff to another PR. It will likely just trade one set of readability concerns for another unless we do it carefully.